gemrb/core/Scriptable/Scriptable.cpp

```diff
@@ -2227,71 +2271,30 @@ void Movable::FixPosition()
 	Pos.y=Pos.y*12+6;
 }
 
-void Movable::WalkTo(const Point &Des, int distance)
-{
-	Point from;
-
-	// maybe caller should be responsible for this
-	if ((Des.x/16 == Pos.x/16) && (Des.y/12 == Pos.y/12)) {
-		ClearPath();
-		return;
-	}
+void Movable::WalkTo(const Point &Des, int distance) {
+	Log(DEBUG, "WalkToWIP", "curTime = %d, prevTicks = %d", Ticks, this->prevTicks);
+	// The rate limiting stuff is probably not optimal, since it requires lots of pathfinder calls
+	// An incremental pathfinding algorithm would be preferable, but LOS checks are slow
+	// and not pre-cachable, while incremental pathfinding requires quite a lot of checks
+	if (this->Ticks > this->prevTicks) { // Rate limiting
```

Comment:
Often it is cleaner to return early, since there's no else block. Allows for a more shallow indentation/nesting of the subsequent code.

But of course some wouldn't agree, mainly proponents of a single exit point.